mediachain/reader/api.py

```diff
@@ -56,12 +56,24 @@ def with_fallback():
     return with_thumb
 
 
+def merge(a, b, path=None):
```

Comment:
Yes, did a more comprehensive merging which detects all possible type paths through the tree, e.g. dealing with problematic cases like:

`merge({'a':{'b':{2:{4:5}}}}, {'a':{'b':[1,2,3]}})`

But is sophisticated merging what we want here, or instead just simple replacement of the whole dict, or precise operations that include e.g. updates to lists and deep targeted update operations like mongo?

Will generalize my type checking / merging code for possible use by the rest of the system (as appropriate) next week.